Update reporter.py to add download functionality for apple ids with multiple accounts

Felevo opened the pull request:
Hi! We struggled with this so I thought I my recomposition might be helpful for others as well.

This new version fixes issues with apple ids that have multiple accounts.
Before, it was not possible to request ```getVendors()``` and ```getReports()``` etc. for apple ids with multiple accounts because the account number was passed as ```a=self.account```. This is no longer valid syntax. Instead, apple wants the account number to be part of Reporter.properties. Equivalent to this, one can pass the account number in the ```data``` object.

Cheers

Felevo committed:
Update reporter.py

chason commented:
Hi Felix!

Thanks again for working on this. If you fix the failing test and put the whitespace lines that got removed back in, I'll be happy to merge this.

Felevo committed:
Added missing whitespace

Felevo commented:
Hi chason,

Not a problem. I included the whitespaces again. However, for the checks to pass the ```test_worker.py``` file needs to be updated as well. The test ```test_account_number_is_passed``` is expecting the account number to be passed as an argument, which I would like to change in the open PR, because it is no longer compliant with the Reporter logic. I will send another PR for the test-file as well.

Best,

Felix

Felevo committed:
changed parameter name for account

it used to be account, but this meant that the ```test_reporter.py``` would fail
